### ISSUE TITLE
HTML widget doesn't keep string literals or links

### ISSUE THREAD

**Issue by tonyluo98** [jupyter-widgets/ipywidgets#2620]
HTML widget value can not process literals such as CRLF, /r/n,  and also if a string contains a link, the widget will just display the link in text format and isn't a clickable link.

For context, I retrieve a string from a website. This string looks like : 
     Hello /r/n This is example string /r/n https://github.com/jupyter-widgets/ipywidgets

This should display like : 
     Hello
     This is example string
     https://github.com/jupyter-widgets/ipywidgets              < --------- (link is pressable and redirects)

But rather displays like : 
     Hello This is example string https://github.com/jupyter-widgets/ipywidgets

In other words, python code print() will display in the format it should be , but display won't.

Is there a way around this?

**Comment by jasongrout** [comment]
The HTML widget just asks the browser to display the content as HTML, nothing more. From what you are describing, it sounds like it is working exactly as intended. If you want formatting such as line breaks in an HTML widget, use the HTML `<br>`, or wrap your content in `<pre>` tags. If you want links, use the HTML `<a>` element.

**Comment by tonyluo98** [comment]
Is there any widgets available that display text that would also display such formatting? As the strings are received through a API therefore it would be more convenient to display through a widget rather than to parse the string first then display via a widget?

**Comment by jasongrout** [comment]
You could print it to an output widget. Then it would use the same text parsing engine that normal output does.

```python
from ipywidgets import Output
out = Output()

with out:
    print('my text')

display(out)
```

**Comment by jasongrout** [comment]
See https://ipywidgets.readthedocs.io/en/latest/examples/Output%20Widget.html

**Comment by jasongrout** [comment]
Closing as answered. Please post again if you want to continue the discussion. Thanks!
